core/src/main/java/org/infinispan/interceptors/impl/InvocationContextInterceptor.java

```diff
@@ -114,7 +110,6 @@ public void init(TransactionManager tm, ComponentRegistry componentRegistry, Tra
          }
       }
 
-      invocationContextContainer.setThreadLocal(ctx);
```

Comment:
+1, the thread-local was causing a performance bottleneck in SPEC (at least in 5.2.x used by EAP 6)